src/dropbox.ml

```diff
@@ -590,6 +612,62 @@ module Make(Client: Cohttp_lwt.Client) = struct
     Client.post ~headers:(headers t) u
     >>= check_errors_404 media_of_response
 
+  let stream_files_put t ?(locale="") ?(overwrite=true) ?(parent_rev="")
+                       ?(autorename=true) fn len stream =
+    let headers = headers t in
+    (* let headers = Cohttp.Header.add (headers t)
+      "Content-Length" (string_of_int len) in *)
+    let u =
+      Uri.of_string("https://api-content.dropbox.com/1/files_put/auto/" ^ fn) in
+    let q = [("overwrite", [string_of_bool overwrite]);
+             ("autorename", [string_of_bool autorename])] in
+    let q = if locale <> "" then ("locale", [locale]) :: q else q in
+    let q = if parent_rev <> "" then ("parent_rev",[parent_rev]) :: q else q in
+    let u = Uri.with_query u q in
+    Client.put ~headers ~body:(Cohttp_lwt_body.of_stream stream) u
+    >>= check_errors >>= fun (_, body) -> Cohttp_lwt_body.to_string body
+    >>= fun body -> return(Json.metadata_of_string body)
+
+  let cohttp_body_files_put t ?(locale="") ?(overwrite=true) ?(parent_rev="")
+                            ?(autorename=true) fn len stream =
```

Comment:
The duplication of code between this function and `stream_files_put` is not nice.